ui/chapter/End.tsx

```diff
@@ -20,9 +22,13 @@ export default function End({
 }) {
   const t = useTranslations(lang)
   const modals = useModalContext()
+  const { account } = useAuthContext()
   const saveAndReturn = useSaveAndReturn()
+  const { chapterId, lessonId } = usePathData()
 
-  const { account } = useAuthContext()
+  const chapterLessons = lessons?.[chapterId]
+  const lesson = chapterLessons?.[lessonId]?.metadata ?? null
+  const currentLessonKey = lesson?.key ?? 'CH1INT1'
```

Comment:
shouldn't we change the CH1INT1 to CH1OUT2 so that if we are unable to fetch the current lesson data the feedback form is at least displayed in chapter 1?

Although upon testing I was unable to find a scenario in which the currentLessonKey was using 'CH1INT1'.

Reply:
this would occur if we decided to put the `end.tsx` component into `intro-1`. While this would never happen perhaps there is a better way to handle lesson being undefined

Reply:
The caller of the End.tsx component must always be a lesson with a defined lessonKey.

So if the currentLessonKey is `undefined` in some cases, it's best to pass a console error to report the discrepancy.

Reply:
I was thinking about,

What if the caller component passes its lesson key to the **End.tsx** component?
This IMO will be a better approach because:

1. The parent will directly inform the child component about its lesson key identity. This eliminates the need for checking the `lessonKey` identity by the child component.
2. It will make the **End.tsx** component more particular in its purpose.
3. It will provide a single source of truth for critical information such as lessonKey.

This shall be a better approach, in general, for all the child components.

Reply:
Ok! I feel like that change may be a bit outside the scope of this PR but I think it is a good possible issue to make into a new PR